ttnn/cpp/ttnn/operations/moreh/moreh_sgd/moreh_sgd.hpp

```diff
@@ -0,0 +1,36 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ttnn/decorators.hpp"
+#include "ttnn/operations/core/compute_kernel/compute_kernel_config.hpp"
+
+namespace ttnn::operations::moreh::moreh_sgd {
+struct MorehSgd {
+    static std::vector<std::optional<Tensor>> invoke(
+        const Tensor& param_in,
+        const Tensor& grad,
+        const std::optional<const Tensor> momentum_buffer_in,
+        const std::optional<const Tensor> param_out,
+        const std::optional<const Tensor> momentum_buffer_out,
+        float lr,
+        float momentum,
+        float dampening,
+        float weight_decay,
+        bool nesterov,
+        bool momentum_initialized,
+        const std::optional<MemoryConfig>& param_out_memory_config,
+        const std::optional<MemoryConfig>& momentum_buffer_out_memory_config,
+        const DeviceComputeKernelConfig compute_kernel_config);
```

Comment:
The concrete compute_kernel_config here seems incorrect. It should be std::optional throughout until the primitive is invoked.

Reply:
resolved in [dc2ce28](https://github.com/tenstorrent/tt-metal/pull/12378/commits/dc2ce28ff17ddff2b33f87920282e226d4fec218)